code/game/gamemodes/game_mode.dm

```diff
@@ -603,3 +606,120 @@
+//Keeps track of all silicon members //
+///////////////////////////////////////
+/datum/game_mode/proc/get_all_silicon()
+	. = list()
+	for(var/mob/living/silicon/player in GLOB.mob_list)
+		if(player.mind && (player.mind.assigned_role in GLOB.nonhuman_positions))
+			. |= player.mind
+
+/datum/game_mode/proc/generate_credit_text()
+		var/list/round_credits = list()
+		var/len_before_addition
+
+		// HEADS OF STAFF
+		round_credits += "<center><h1>The Glorious Command Staff:</h1>"
+		len_before_addition = round_credits.len
+		for(var/datum/mind/current in SSticker.mode.get_all_by_department(GLOB.command_positions))
+				round_credits += "<center><h2>[current.name] as the [current.assigned_role]</h2>"
+		if(round_credits.len == len_before_addition)
+				round_credits += list("<center><h2>A serious bureaucratic error has occurred!</h2>", "<center><h2>No one was in charge of the crew!</h2>")
+		round_credits += "<br>"
+
+		// SILICONS
+		round_credits += "<center><h1>The Silicon \"Intelligences\":</h1>"
+		len_before_addition = round_credits.len
+		for(var/datum/mind/current in SSticker.mode.get_all_silicon())
+				round_credits += "<center><h2>[current.name] as the [current.assigned_role]</h2>"
+		if(round_credits.len == len_before_addition)
+				round_credits += list("<center><h2>[station_name()] had no silicon helpers!</h2>", "<center><h2>Not a single door was opened today!</h2>")
+		round_credits += "<br>"
+
+		// SECURITY
+		round_credits += "<center><h1>The Brave Security Officers:</h1>"
+		len_before_addition = round_credits.len
+		for(var/datum/mind/current in SSticker.mode.get_all_by_department(GLOB.security_positions))
+				round_credits += "<center><h2>[current.name] as the [current.assigned_role]</h2>"
+		if(round_credits.len == len_before_addition)
+				round_credits += list("<center><h2>[station_name()] has fallen to Communism!</h2>", "<center><h2>No one was there to protect the crew!</h2>")
+		round_credits += "<br>"
+
+		// MEDICAL
+		round_credits += "<center><h1>The Wise Medical Department:</h1>"
+		len_before_addition = round_credits.len
+		for(var/datum/mind/current in SSticker.mode.get_all_by_department(GLOB.medical_positions))
+				round_credits += "<center><h2>[current.name] as the [current.assigned_role]</h2>"
+		if(round_credits.len == len_before_addition)
+				round_credits += list("<center><h2>Healthcare was not included!</h2>", "<center><h2>There were no doctors today!</h2>")
+		round_credits += "<br>"
+
+		// ENGINEERING
+		round_credits += "<center><h1>The Industrious Engineers:</h1>"
+		len_before_addition = round_credits.len
+		for(var/datum/mind/current in SSticker.mode.get_all_by_department(GLOB.engineering_positions))
+				round_credits += "<center><h2>[current.name] as the [current.assigned_role]</h2>"
+		if(round_credits.len == len_before_addition)
+				round_credits += list("<center><h2>[station_name()] probably did not last long!</h2>", "<center><h2>No one was holding the station together!</h2>")

```

Comment:
What if the captain or an assistant or something was "holding the station together"?

/nitpick